backend/editor/entries.py

```diff
@@ -40,28 +40,31 @@ def create_node(self, label, entry, main_language_code):
         """
         Helper function used for creating a node with given id and label
         """
-        # Normalizing new Node ID
-        normalised_entry = normalizer.normalizing(entry, main_language_code)
-
+        params = {"id": entry}
         query = [f"""CREATE (n:{self.project_name}:{label})\n"""]
-        params = {"id": normalised_entry}
 
         # Build all basic keys of a node
         if (label == "ENTRY"):
-            canonical_tag = normalised_entry.split(":", 1)[1]
-            query.append(f""" SET n.main_language = $main_language_code """) # Required for only an entry
+            # Normalizing new canonical tag
+            language_code, canonical_tag = entry.split(":", 1)
+            normalised_canonical_tag = normalizer.normalizing(canonical_tag, main_language_code)
+            
+            # Reconstructing and updation of node ID
+            params["id"] = language_code + ':' + normalised_canonical_tag
             params["main_language_code"] = main_language_code
+
+            query.append(f""" SET n.main_language = $main_language_code """) # Required for only an entry
         else:

```

Comment:
FYI:

If we want to support stopwords and synonyms there might be more work to be done:
from docs/references/taxonomy-parser.md:
```
2. store each block of *stopwords line* in a node with id `stopwords:IDX`
   and label STOPWORDS,
   where `IDX` is the number of preceding *stopwords lines* in the file
3. store each block of *global synonyms line*
   in a node with id `synonyms:IDX`
   and label SYNONYMS,
   where `IDX` is the number of preceding *global synonyms lines* 
   in the file
```
and it's not the responsability of the frontend to compute this IDX part.